### ISSUE TITLE
Errors running command file BuildBoot.cmd in Nitra solution

### ISSUE THREAD

**Issue by rcammarere** [JetBrains/Nitra#31]
I downloaded Nitra and am very anxious to start working with it. However, when I ran BuildBoot.cmd, I got a series of error messages similar to this:
Internal\ExtensibleRuleParseTreeConstructor.n(74,36,74,39): error : typing fail
s on accessing member Add in the ? type [C:\NitraBase\Nitra\Boot1\Nitra.Runtime\Boot.Nitra.Runtime.npro
j]

Also, I get this error when building in VS . . . The referenced assembly “C:\NitraBase\Nitra\Nitra.Runtime\bin\Debug\Nitra.Runtime.dll” was not found. If this assembly is produced by another one of your projects, please make sure to build that project before building this one. Nitra.TestsLauncher

**Comment by someone-with-default-username** [comment]
Have you downloaded and installed Nemerle or built it from sources? What branch are you trying to build?

**Comment by someone-with-default-username** [comment]
Seems that LightList type presents in Nitra.Runtime from master branch (it is quite old), but now it is moved to Nemerle.dll. I'm going to refresh sources in master.

**Comment by rcammarere** [comment]
Yes, I did install Nemerle (the latest version)

**Comment by rcammarere** [comment]
I went here . . . https://github.com/JetBrains/Nitra  and clicked on "Download ZIP".  That's what I'm working with now.  Also I navigated to here . . http://nemerle.org/Downloads  and installed version 4.0 v1.2.445.0 (2015-04-15).  You said that you were going to refresh sources in master.  Should I remove what I've got and download the ZIP file for Nitra again?

Thanks,
Renee

**Comment by rcammarere** [comment]
Rather than downloading the ZIP file.  I decided to download GitHub and create a repository for the master branch.  However, this is the first time I'm using this.  In an attempt to clone Nitra, I have this folder on my machine . . .  C:\Users\Renee Cammarere\Documents\GitHub\https---github.com-JetBrains-Nitra.git.  I'm not sure what these files are.  How do I clone Nitra on my machine?  I'm assuming that the ZIP file is not up to date, correct?

**Comment by VladD2** [comment]
> Also I navigated to here . . http://nemerle.org/Downloads and installed version 4.0 v1.2.445.0 . 

Try to use Nemerle v1.2.439.0 (2015-03-21)

**Comment by VladD2** [comment]
> How do I clone Nitra on my machine?

1. You are trying to clone Nemerle repository, instead of Nitra repository. You have already installed Nemerle. You sould't so clone the repository. Use git@github.com:JetBrains/Nitra.git to clone Nitra repository.
2. In Destination field set "C:\NitraBase". Make sure that the "C:\NitraBase" exists before start clone.

**Comment by rcammarere** [comment]
I just uninstalled version 4.0 v1.2.445.0 of Nemerle and installed version v1.2.439.0 (2015-03-21).  When I ran BuildBoot.cmd, I only got 1 error and 1 warning this time, instead of 20 errors (which is what I got before).  However, when I went into the Nitra solution (which I got from the ZIP file) using VS and did a Rebuild, I still get the same errors as before 
![case4195attempttoclonenitraerrors](https://cloud.githubusercontent.com/assets/11961367/7188700/b7ffe352-e447-11e4-9095-a64b8b5e4be9.jpg)

**Comment by rcammarere** [comment]
By the way, I tried again to clone Nitra, this time using the right address, and this is what happened . . .
![case4195attempttoclonenitraerrors2](https://cloud.githubusercontent.com/assets/11961367/7188895/f3f33d40-e448-11e4-83d3-7a0b6cfeb6c1.jpg)
If I could get the Nitra solution that I downloaded from the ZIP file to build error-free, I wouldn't need to clone.  Thanks so much for your help.  I'm hoping to get this resolved soon.  My boss keeps asking me about it :-)

**Comment by rcammarere** [comment]
I can't get this to build without errors.  Can I call one of your developers, or have someone give me a call?  Thanks.

**Comment by VladD2** [comment]
> By the way, I tried again to clone Nitra, this time using the right address, and this is what happened . . .

I think, you have not installed (or incorrectly installed) [git](http://git-scm.com/downloads).

PS

I fix 'master'. Now it sould build with the latest Nemerle version.

PPS

If you simply want to try Nitra, you can download the installer (Nitra.Setup.msi) [here](https://confluence.jetbrains.com/display/Nitra/Install).

**Comment by rcammarere** [comment]
I'm sure you're right.  I keep reading the documentation, but I'm missing something.  I just want to try Nitra, so I'll go to the link that you indicated.  Thanks.

**Comment by rcammarere** [comment]
I was able to get it installed and running, and the GUI comes up.  However, I find that I am not able to test it because I must be missing something.  I may be making the wrong assumptions. How do I test Nitra.Visualizer.exe using the samples?  I noticed that it is looking for an "nsln" file and I can load one of the  nsln files from the Tests\Visualizer folder (from when I downloaded the Nitra solution before), but then I get errors because it can't find certain dll's.

In other words, how do I test it once I have this GUI?
![case4195attempttousenitra](https://cloud.githubusercontent.com/assets/11961367/7210022/2d1a457c-e51c-11e4-8b9d-f7e9e3dcfa86.jpg)
The reason you see NitraGrammar in there is because I brought in Nitra.Grammar.dll (just testing)
Thanks,
Renee

**Comment by rcammarere** [comment]
That latest update that you did to 'master' did the trick.  I had no errors when I ran BuildBoot.cmd this time. Thanks :-)
